src/overlays/actors/ovl_En_Talk_Gibud/z_en_talk_gibud.c

```diff
@@ -28,166 +83,1057 @@ const ActorInit En_Talk_Gibud_InitVars = {
+    if (EnTalkGibud_PlayerInRangeWithCorrectState(this, globalCtx) && Actor_IsActorFacingLink(&this->actor, 0x38E3)) {
+        if (this->grabWaitTimer == 0 && this->actor.xzDistToPlayer <= 45.0f) {
+            player->actor.freezeTimer = 0;
+            if (gSaveContext.playerForm == PLAYER_FORM_GORON || gSaveContext.playerForm == PLAYER_FORM_DEKU) {
+                // If the Gibdo/Redead tries to grab Goron or Deku Link, it will fail to
+                // do so. It will appear to take damage and shake its head side-to-side.
+                EnTalkGibud_SetupGrabFail(this);
+            } else if (globalCtx->grabPlayer(globalCtx, player)) {
+                EnTalkGibud_SetupGrab(this);
+            }
+        } else {
+            if (this->playerStunWaitTimer == 0) {
+                player->actor.freezeTimer = 40;
+                this->playerStunWaitTimer = 60;
+                func_8013ECE0(this->actor.xzDistToPlayer, 255, 20, 150);
+                func_80123E90(globalCtx, &this->actor);
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_AIM);
+            } else {
+                this->playerStunWaitTimer--;
+            }
+        }
+    } else if (this->grabWaitTimer == 0 && this->actor.xzDistToPlayer <= 45.0f) {
+        EnTalkGibud_SetupWalkToHome(this);
+    } else if (EnTalkGibud_PlayerOutOfRange(this, globalCtx)) {
+        EnTalkGibud_SetupWalkToHome(this);
+    }
+    if (this->grabWaitTimer > 0) {
+        this->grabWaitTimer--;
+    }
+
+    if (Animation_OnFrame(&this->skelAnime, 10.0f) || Animation_OnFrame(&this->skelAnime, 22.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_WALK);
+    } else if (!(globalCtx->gameplayFrames & 0x5F)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_CRY);
+    }
+}
+
+void EnTalkGibud_SetupGrab(EnTalkGibud* this) {
+    func_800BDC5C(&this->skelAnime, sAnimations, EN_TALK_GIBUD_ANIMATION_GRAB_START);
+    this->grabDamageTimer = 0;
+    this->actor.flags &= ~1;
+    this->grabState = EN_TALK_GIBUD_GRAB_START;
+    this->actionFunc = EnTalkGibud_Grab;
+}
+
+void EnTalkGibud_Grab(EnTalkGibud* this, GlobalContext* globalCtx) {
+    // This function needs to have two different temps for Player to match,
+    // but you don't have to necessarily use them both. This is just the most
+    // likely scenario; they got an Actor* pointer in the first temp, then
+    // casted it to Player* in the second temp.
+    Actor* playerActor = &GET_PLAYER(globalCtx)->actor;
+    Player* player = (Player*)playerActor;
+    s32 inPositionToAttack;
+    u16 damageSfxId;
+
+    switch (this->grabState) {
+        case EN_TALK_GIBUD_GRAB_START:
+            inPositionToAttack = EnTalkGibud_MoveToIdealGrabPositionAndRotation(this, globalCtx);
+            // Needs to have this "== true" to match.
+            if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame) && inPositionToAttack == true) {
```

Comment:
```suggestion
            if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame) && inPositionToAttack == true) {
```
I don't think this comment is needed imo, there's not any alternative when you have `== 1`, and it's pretty clear what it means